optimum/exporters/neuron/__main__.py

```diff
@@ -83,10 +92,17 @@ def infer_task(task: str, model_name_or_path: str) -> str:
 
 def normalize_input_shapes(task: str, args: argparse.Namespace) -> Dict[str, int]:
     if task == "stable-diffusion":
-        mandatory_shapes = {
-            name: getattr(args, name, None)
-            for name in getattr(inspect.getfullargspec(build_stable_diffusion_components_mandatory_shapes), "args")
-        }
+        args = vars(args) if isinstance(args, argparse.Namespace) else args
+        mandatory_axes = set(
+            getattr(inspect.getfullargspec(build_stable_diffusion_components_mandatory_shapes), "args")
+        )
+        # Remove `sequence_length` as diffusers will pad it to the max anyway.
+        mandatory_axes.remove("sequence_length")
+        if not mandatory_axes.issubset(set(args.keys())):
+            raise AttributeError(
+                f"Shape of {mandatory_axes} are mandatory for neuron compilation, while {mandatory_axes.difference(args.keys())} are not given."
+            )
+        mandatory_shapes = {name: args[name] for name in mandatory_axes}
```

Comment:
I feel like instead of branching a lot in methods like that, we should have stable diffusion specific functions for that and just call them here.

From my experience with `optimum.exporters.onnx`, adding Stable Diffusion support added a lot of "complexity" in the code because we always need to do something different for these kinds of models. WDTY?

Reply:
done for this function, maybe we can also separate the main_export of sd, but let's do it when I add other tasks (the moment that the complexity increases so that I can have a better picture).